### PR TITLE
Refactor service provider

### PR DESCRIPTION
**Pull request by GrahamCampbell**
This should be easier to read now, and should be cleaner for us to modify this to support guzzle 5 too later on.

### CODE REVIEW

File: src/BugsnagServiceProvider.php [hunk 1]
```diff
@@ -28,27 +29,52 @@ class BugsnagServiceProvider extends ServiceProvider
      */
     public function boot()
     {
+        $this->app->call([$this, 'setupConfig']);
```
Comment (vinkla) [inline src/BugsnagServiceProvider.php]: I'm curious, why are you calling them like this instead of `$this->setupConfig()`?

Comment (GrahamCampbell) [inline src/BugsnagServiceProvider.php]: Allows automatic injection of the application instance.

Comment (GrahamCampbell) [inline src/BugsnagServiceProvider.php]: Just makes for shorter, more readable code inside the setupConfig method.

Comment (vinkla) [inline src/BugsnagServiceProvider.php]: Oh, I see.

**Pull request merged**